Source/WebCore/page/Page.cpp

```diff
@@ -1938,6 +1938,15 @@ void Page::didCompleteRenderingUpdateDisplay()
     m_inspectorController->didComposite(mainFrame());
 }
 
+void Page::didCompleteRenderingFrame()
```

Comment:
It would be ideal to all these updating per `WindowEventLoop` object. e.g. cross-site contents in a single Page may not share EventLoop object (this is especially true once site isolation is implemented).

Reply:
Agreed, but I'd rather do that later. It should be pretty clean; the Page `updateRendering` code is in a good state.